src/data.rs

```diff
@@ -453,6 +468,73 @@ pub fn decode_block_cid_ask_message(msg: Vec<u8>) -> Option<(i128, Option<Cid>)>
 	}
 }
 
+async fn fetch_cell_from_ipfs(
+	ipfs: &Ipfs<DefaultParams>,
+	peers: Vec<PeerId>,
+	block_number: u64,
+	position: &DataCell,
+) -> Result<DataCell> {
+	let record_key = Key::from(position.reference().as_bytes().to_vec());
+
+	log::trace!("Getting DHT record for reference {}", position.reference());
+	let cid = ipfs
+		.get_record(record_key, ipfs_embed::Quorum::One)
+		.await
+		.map(|record| record[0].record.value.to_vec())
+		.and_then(|value| Cid::try_from(value).context("Invalid CID value"))?;
+
+	log::trace!("Fetching IPFS block for CID {}", cid);
+	ipfs.fetch(&cid, peers).await.and_then(|block| {
+		Ok(DataCell {
+			block: block_number,
+			row: position.row,
+			col: position.col,
+			proof: block.data().to_vec(),
+			data: block.data()[48..].to_vec(),
+		})
+	})
+}
+
+pub async fn fetch_cells_from_ipfs(
```

Comment:
We should change (or remove) the `ipfs_priority_get_cells` function in the LC, to make sure we have uniform approach to data retrieval.